src/lib/coerce.js

```diff
@@ -12,21 +12,38 @@ var nestedProperty = require('./nested_property');
 var counterRegex = require('./regex').counter;
 var modHalf = require('./mod').modHalf;
 var isArrayOrTypedArray = require('./array').isArrayOrTypedArray;
+var isTypedArraySpec = require('./array').isTypedArraySpec;
+var decodeTypedArraySpec = require('./array').decodeTypedArraySpec;
 
 exports.valObjectMeta = {
     data_array: {
         // You can use *dflt=[] to force said array to exist though.
         description: [
             'An {array} of data.',
-            'The value MUST be an {array}, or we ignore it.',
-            'Note that typed arrays (e.g. Float32Array) are supported.'
+            'The value must represent an {array} or it will be ignored,',
+            'but this array can be provided in several forms:',
+            '(1) a regular {array} object',
+            '(2) a typed array (e.g. Float32Array)',
+            '(3) an object with keys dtype, vals, and optionally shape.',
+            'In this 3rd form, dtype is one of *i1*, *u1*, *c1*,',
+            '*i2*, *u2*, *i4*, *u4*, *f4*, or *f8*.',
+            'vals is either a base64-encoded string or the ArrayBuffer of',
```

Comment:
I am tempted to rename `vals` to `buffer`.
@alexcjohnson What do you think?

Reply:
IMHO using `data` would conflict with figure data.
What about `bdata` which points to both bit64 and buffer?
In that case all these keys would have 5 characters i.e. `shape`, `dtype` and `bdata` which is nice.

Reply:
I don't personally find `data` to be problematic, but I see your point. I could go for `bdata`, to me the `b` means "binary" which also works for both the base64 and ArrayBuffer cases.